src/cargo/sources/registry/remote.rs

```diff
@@ -33,8 +42,80 @@ impl<'cfg> RemoteRegistry<'cfg> {
             cache_path: config.registry_cache_path().join(name),
             source_id: source_id.clone(),
             config: config,
-            handle: None,
+            tree: RefCell::new(None),
+            handle: LazyCell::new(),
+            repo: LazyCell::new(),
+            head: Cell::new(None),
+        }
+    }
+
+    fn easy(&self) -> CargoResult<&RefCell<Easy>> {
+        if let Some(handle) = self.handle.borrow() {
+            return Ok(handle)
+        }
+        let handle = ops::http_handle(self.config)?;
+        self.handle.fill(RefCell::new(handle)).ok().unwrap();
+        Ok(self.handle.borrow().unwrap())
+    }
```

Comment:
Looks like `easy` and `repo` could use `LazyCell::get_or_try_init` instead of manually unwrapping things.